README
Yealink V6X, V7X firmware T2x,T3x, T4x, W52 and VP530 voice phone support plugin for sipXecs and openUC
Localization: en_EN, ru_RU
(C)2012 SIPLABS Communications (SibTelCom, JSC)
Contact e-mail: info@siplabs.ru
Contact phone: +7-383-363-2111
Contact URL: http://www.siplabs.ru



changelog.txt
RPM V4.6.0 build 1724.b0d0d
    Date:   Fri Sep 27 13:32:21 2013 +0700
	FIX: linekeys/memorykey BLF in SIP-T[23]X
    Date:   Thu Sep 26 11:21:15 2013 +0700
	FIX: BLF keys for SIP-T[1-3]X phones were not assigned and not visible in sipXconfig WEB UI
    Date:   Thu Sep 26 00:53:39 2013 +0700
	ADD: Directed Pickup Code to phone defaults
    Date:   Thu Sep 26 00:17:09 2013 +0700
	Refactoring: audio codecs choosing via multi-enum setting

RPM V4.6.0 build 1724.b0d0d
    Date:   Tue Sep 24 11:53:51 2013 +0700
	FIX: BLF present in sipXecs WEB UI, but was not present in MAC.cfg file
    Date:   Tue Sep 24 02:14:03 2013 +0700
	FIX: linekey,programableley defaults
    Date:   Mon Sep 23 16:43:01 2013 +0700
	Add: tablevoew support for linekey.
	Add: dynamic defaults for linekey and programablekey collections.
	Fix: programable type for programable keys.
    Date:   Thu Sep 19 15:34:59 2013 +0700
	Added: table-view support for programblekey, alert-info and autop sources
    Date:   Tue Sep 17 13:05:07 2013 +0700
	Refactoring: Remote Phonebooks order and set selection with multi-enum

RPM V4.6.0 build 1685.74e33
    Date:   Thu Sep 12 12:15:56 2013 +0700
	FIX: Line index for V71 and V70 FW.
    Date:   Wed Sep 11 16:45:46 2013 +0700
	Add: local contacts uploading support via sipXecs device files management UI.
    Date:   Tue Sep 10 15:39:22 2013 +0700
	Update README
    Date:   Tue Sep 10 14:24:20 2013 +0700
	FIX: LDAP password is visible in WEB UI.
	ADD: LDAP "Enable" default value dependent of LDAP configured in UC system.
    Date:   Tue Sep 10 13:21:12 2013 +0700
	Added: DirectCallPickup code as a default value for DSS 'extension' and 'pickup_value' values
    Date:   Tue Sep 10 11:50:01 2013 +0700
	Add: LDAP connection default settings from first system LDAP connection
    Date:   Mon Sep 9 11:01:01 2013 +0700
	FIX: label for linekey.X.extension
    Date:   Mon Sep 9 10:56:14 2013 +0700
	Added: BLF or SpeedDial type of button set from speeddial.Button object.
	FIX: default values for lines of left-sided button for SIP-T4X phones.
	FIX: xml_phonebook parameter is unused on SIP-T4X
	FIX: lines are 1-based sequences - not 0-based
    Date:   Fri Sep 6 16:14:40 2013 +0700
	FIXED: FW file name for SIp-T46 was incorrect.
	Added: DSS keys 4-6 for SIp-T41,SIP-T42G and DSS keys 6-9 for SIP-T46G assigned as "Line" keys. Other keys are assigned to N/A by default and values for other keys are assigned as speeddial buttons values.
    Date:   Fri Sep 6 12:36:49 2013 +0700
	Added: JAVA DSS keys generation from bean propertie maxDSSKeyCount and maxLineCount(for SIP-T[1-3]X models only)
    Date:   Wed Sep 4 14:49:19 2013 +0700
	Refactoring: Added YealinkEnumSetter abstract class, which implement most common functionality for enumerators filling from diffrent sources.
	Refactoring: Moved from type.xml DKType labels to *.properties.
	Add: maxDSSKeyCount propertie for all models, deprecated memoryKeyCount propertie.
    Date:   Tue Sep 3 14:10:02 2013 +0700
	Add: DSS key types for V71 FW and SIP-T4X phones
    Date:   Thu Aug 22 12:40:37 2013 +0700
	Add: multiple lang(+/-)Language.txt files uploading.
    Date:   Mon Aug 19 20:18:02 2013 +0700
	FIX: SoftKeys configuration was filtered out by model filter
	FIX: SoftKeys default values
	FIX: Default version now set to V7X
	FIX: Bug if phone was not created, phone group edition causes an exception - unable to open model file 'phone.xml'

RPM V4.6.0 build 1622.e2f62
    Date:   Thu Aug 15 20:17:10 2013 +0700
	ADD: MAC address prefix on phone creation
	ADD: WallPapers and ScreenSavers uploading for color LCD phones
	ADD: features support for phone beans
	ADD: filters for settings difference between different models

RPM v.4.6.0 build 1621.a4224
    Date:   Wed Aug 14 07:03:08 2013 +0700
	ADD: RingTones uploading and selection.
	ADD: PhoneBook selection (only renames phonebook)
    Date:   Tue Aug 13 13:20:04 2013 +0700
	FIXED: LCD logo
	ADED: multiple phonebook support

RPM 4.6.0 build 1580.e4c62
    Date:   Tue Aug 6 15:45:45 2013 +0700
	FIX: properties for lang file url
    Date:   Tue Aug 6 15:18:59 2013 +0700
	Common settings are ready.
    Date:   Fri Aug 2 17:43:35 2013 +0700
	Added SIP-T46G, SIP-T42G, SIP-T41, W52, VP530 models, finished page Action URL
    Date:   Thu Aug 1 17:09:31 2013 +0700
	Wiped out all unsupported settings. Compleated Upgrade section.
    Date:   Wed Jul 31 16:05:29 2013 +0700
	Refactor: universal support for 2 level groups
    Date:   Fri Jul 12 13:15:10 2013 +0700
	Compleated section Preference, FIXED: DNS defaults
    Date:   Tue Jul 9 11:30:18 2013 +0700


src/org/sipfoundry/sipxconfig/phone/yealink/YealinkConstants.java
/*
 * Copyright (C) 2013 SibTelCom, JSC., certain elements licensed under a Contributor Agreement.
 * Author: Konstantin S. Vishnivetsky
 * E-mail: info@siplabs.ru
 * Contributors retain copyright to elements licensed under a Contributor Agreement.
 * Licensed to the User under the LGPL license.
 *
*/

package org.sipfoundry.sipxconfig.phone.yealink;

public class YealinkConstants {

    public static final String MIME_TYPE_PLAIN = "text/plain";
    public static final String MIME_TYPE_XML = "text/xml";

    public static final String XML_CONTACT_DATA = "directory.xml";
    public static final String WEB_ITEMS_LEVEL = "webitemslevel.cfg";
    public static final String VENDOR = "Yealink";
    public static final String MAC_PREFIX = "001565";

    public static final String PKTYPES_V6X = "0,2,5,6,7,8,13,22,28,29,30,31,32,33,39,43,44,45,46,47";
    public static final String PKTYPES_V70 = "0,2,5,6,7,8,9,13,14,22,23,27,28,29,30,32,33,43,44,45,46,47,48,49,50";
    public static final String DKTYPES_V70 = "0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,27,34,35,38,39,40,41,42,45,46";
    public static final String DKTYPES_V71 = "0,11,12,13,14,15,16,17,18,22,23,24,25,27,34,35,38,40";

    // Line specific settings used in /etc/yealinkPhone/line_XX.xml
    public static final String USER_ID_V6X_SETTING = "account/UserName";
    public static final String USER_ID_V7X_SETTING = "basic/user_name";
    public static final String AUTH_ID_V6X_SETTING = "account/AuthName";
    public static final String AUTH_ID_V7X_SETTING = "basic/auth_name";
    public static final String DISPLAY_NAME_V6X_SETTING = "account/DisplayName";
    public static final String DISPLAY_NAME_V7X_SETTING = "basic/display_name";
    public static final String PASSWORD_V6X_SETTING = "account/password";
    public static final String PASSWORD_V7X_SETTING = "basic/password";
    public static final String REGISTRATION_SERVER_HOST_V6X_SETTING = "account/SIPServerHost";
    public static final String REGISTRATION_SERVER_HOST_V7X_SETTING = "basic/sip_server_host";
    public static final String REGISTRATION_SERVER_PORT_V6X_SETTING = "account/SIPServerPort";
    public static final String REGISTRATION_SERVER_PORT_V7X_SETTING = "basic/sip_server_port";
    public static final String OUTBOUND_HOST_V6X_SETTING = "account/OutboundHost";
    public static final String OUTBOUND_HOST_V7X_SETTING = "basic/outbound_host";
    public static final String OUTBOUND_PORT_V6X_SETTING = "account/OutboundPort";
    public static final String OUTBOUND_PORT_V7X_SETTING = "basic/outbound_port";
    public static final String BACKUP_OUTBOUND_HOST_V6X_SETTING = "account/BackOutboundHost";
    public static final String BACKUP_OUTBOUND_HOST_V7X_SETTING = "basic/backup_outbound_host";
    public static final String BACKUP_OUTBOUND_PORT_V6X_SETTING = "account/BackOutboundPort";
    public static final String BACKUP_OUTBOUND_PORT_V7X_SETTING = "basic/backup_outbound_port";
    public static final String VOICE_MAIL_NUMBER_V6X_SETTING = "Message/VoiceNumber";
    public static final String VOICE_MAIL_NUMBER_V7X_SETTING = "basic/voice_mail.number";

    // Phone specific settings used in /etc/yealinkPhone/phone_XX.xml
    public static final String DNS_SERVER1_V6X_SETTING = "network-wan/DNS/PrimaryDNS";
    public static final String DNS_SERVER1_V7X_SETTING = "network-wan/DNS/network.primary_dns";
    public static final String DNS_SERVER2_V6X_SETTING = "network-wan/DNS/SecondaryDNS";
    public static final String DNS_SERVER2_V7X_SETTING = "network-wan/DNS/network.secondary_dns";
    public static final String SYSLOG_SERVER_V6X_SETTING = "upgrade/SYSLOG/SyslogdIP";
    public static final String SYSLOG_SERVER_V7X_SETTING = "upgrade/SYSLOG/syslog.server";
    public static final String LOCAL_TIME_ZONE_V6X_SETTING = "preference/Time/TimeZone";
    public static final String LOCAL_TIME_ZONE_V7X_SETTING = "preference/local_time.time_zone";
    public static final String LOCAL_TIME_SERVER1_V6X_SETTING = "preference/Time/TimeServer1";
    public static final String LOCAL_TIME_SERVER1_V7X_SETTING = "preference/local_time.ntp_server1";
    public static final String LOCAL_TIME_SERVER2_V6X_SETTING = "preference/Time/TimeServer2";
    public static final String LOCAL_TIME_SERVER2_V7X_SETTING = "preference/local_time.ntp_server2";
    public static final String XML_BROWSER_SERVER_V6X_SETTING = "xml-browser/ServerIP";
    public static final String XML_BROWSER_SERVER_V7X_SETTING = "features/APISECURITY/push_xml.server";
    public static final String ACTION_URI_LIMIT_IP_V6X_SETTING = "Features/ActionURILimitIP";
    public static final String ACTION_URI_LIMIT_IP_V7X_SETTING = "features/APISECURITY/features.action_uri_limit_ip";
    public static final String DIRECT_CALL_PICKUP_CODE_V7X_SETTING = "features/CALLPICKUP/features.pickup.direct_pickup_code";

    public static final String REMOTE_PHONEBOOK_0_NAME_V6X_SETTING = "contacts/RemotePhoneBook/0/Name";
    public static final String REMOTE_PHONEBOOK_0_URL_V6X_SETTING = "contacts/RemotePhoneBook/0/URL";
    public static final String REMOTE_PHONEBOOK_0_NAME_V7X_SETTING = "contacts/RemotePhoneBook/remote_phonebook.data.1.name";
    public static final String REMOTE_PHONEBOOK_0_URL_V7X_SETTING = "contacts/RemotePhoneBook/remote_phonebook.data.1.url";

    public static final String REMOTE_PHONEBOOK_1_NAME_V6X_SETTING = "contacts/RemotePhoneBook/1/Name";
    public static final String REMOTE_PHONEBOOK_1_URL_V6X_SETTING = "contacts/RemotePhoneBook/1/URL";
    public static final String REMOTE_PHONEBOOK_1_NAME_V7X_SETTING = "contacts/RemotePhoneBook/remote_phonebook.data.2.name";
    public static final String REMOTE_PHONEBOOK_1_URL_V7X_SETTING = "contacts/RemotePhoneBook/remote_phonebook.data.2.url";

    public static final String REMOTE_PHONEBOOK_2_NAME_V6X_SETTING = "contacts/RemotePhoneBook/2/Name";
    public static final String REMOTE_PHONEBOOK_2_URL_V6X_SETTING = "contacts/RemotePhoneBook/2/URL";
    public static final String REMOTE_PHONEBOOK_2_NAME_V7X_SETTING = "contacts/RemotePhoneBook/remote_phonebook.data.3.name";
    public static final String REMOTE_PHONEBOOK_2_URL_V7X_SETTING = "contacts/RemotePhoneBook/remote_phonebook.data.3.url";

    public static final String REMOTE_PHONEBOOK_3_NAME_V6X_SETTING = "contacts/RemotePhoneBook/3/Name";
    public static final String REMOTE_PHONEBOOK_3_URL_V6X_SETTING = "contacts/RemotePhoneBook/3/URL";
    public static final String REMOTE_PHONEBOOK_3_NAME_V7X_SETTING = "contacts/RemotePhoneBook/remote_phonebook.data.4.name";
    public static final String REMOTE_PHONEBOOK_3_URL_V7X_SETTING = "contacts/RemotePhoneBook/remote_phonebook.data.4.url";

    public static final String REMOTE_PHONEBOOK_4_NAME_V6X_SETTING = "contacts/RemotePhoneBook/4/Name";
    public static final String REMOTE_PHONEBOOK_4_URL_V6X_SETTING = "contacts/RemotePhoneBook/4/URL";
    public static final String REMOTE_PHONEBOOK_4_NAME_V7X_SETTING = "contacts/RemotePhoneBook/remote_phonebook.data.5.name";
    public static final String REMOTE_PHONEBOOK_4_URL_V7X_SETTING = "contacts/RemotePhoneBook/remote_phonebook.data.5.url";

    public static final String FIRMWARE_SERVER_ADDRESS_SETTING = "upgrade/firmware/server_ip";
    public static final String FIRMWARE_URL_V6X_SETTING = "upgrade/firmware/url";
    public static final String FIRMWARE_URL_V7X_SETTING = "upgrade/firmware/firmware.url";
    public static final String FIRMWARE_HTTP_URL_SETTING = "upgrade/firmware/http_url";
    public static final String FIRMWARE_NAME_SETTING = "upgrade/firmware/firmware_name";
    public static final String AUTOPROVISIONING_SERVER_URL_V6X_SETTING = "upgrade/autoprovision/strServerURL";
    public static final String AUTOPROVISIONING_SERVER_URL_V7X_SETTING = "upgrade/autoprovision/auto_provision.server.url";
    public static final String AUTOPROVISIONING_SERVER_ADDRESS_V6X_SETTING = "upgrade/autoprovision/server_address";
    public static final String ADVANCED_MUSIC_SERVER_URI_V6X_SETTING = "account/MusicServerUri";
    public static final String ADVANCED_MUSIC_SERVER_URI_V7X_SETTING = "advanced/music_server_uri";
// T2X except T20
    public static final String LOGO_FILE_NAME_V6X_SETTING = "upgrade/Logo/server_address";
    public static final String LOGO_FILE_NAME_V7X_SETTING = "features/GENERAL/lcd_logo.url";

    public static final String DIAL_NOW_URL_V7X_SETTING = "downloads/dialplan_dialnow.url";

    public static final String FEATURE_HDSOUND = "hasHDSound";
    public static final String FEATURE_PHONEBOOK = "hasPhoneBook";
    public static final String FEATURE_RINGTONES = "hasRingTones";
    public static final String FEATURE_WALLPAPERS = "hasWallPapers";
    public static final String FEATURE_SCREENSAVERS = "hasScreenSavers";
    public static final String FEATURE_LANGUAGES = "hasLanguages";

// LDAP
    public static final String LDAP_ENABLE = "contacts/LDAP/ldap.enable";
    public static final String LDAP_HOST = "contacts/LDAP/ldap.host";
    public static final String LDAP_PORT = "contacts/LDAP/ldap.port";
    public static final String LDAP_USER = "contacts/LDAP/ldap.user";
    public static final String LDAP_PASSWORD = "contacts/LDAP/ldap.password";
    public static final String LDAP_BASE = "contacts/LDAP/ldap.base";
}
